src/formatDate.js

```diff
@@ -9,6 +9,67 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const unformattedDate = dateSplitter(date, fromFormat);
+  const separator = toFormat[3];
+  const resArr = [];
+  let res = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'DD') {
+      resArr.push(unformattedDate[0]);
+    }
+
+    if (toFormat[i] === 'MM') {
+      resArr.push(unformattedDate[1]);
+    }
+
+    if (toFormat[i] === 'YYYY') {
+      resArr.push(unformattedDate[2]);
+    }
+
+    if (toFormat[i] === 'YY') {
+      resArr.push(unformattedDate[2] % 100);
+    }
+  }
```

Comment:
The current implementation uses a separate loop for each date part. This could be optimized by using a single loop and a switch statement. This way, you only need to iterate through the `toFormat` array once. Here's a rough idea of how you could do it:

js
for (let i = 0; i < toFormat.length; i++) {
  switch (toFormat[i]) {
    case 'DD':
      resArr.push(unformattedDate[0]);
      break;
    case 'MM':
      resArr.push(unformattedDate[1]);
      break;
    case 'YYYY':
    case 'YY':
      resArr.push(unformattedDate[2]);
      break;
  }
}
```

Please note that in this case, the 'YY' and 'YYYY' cases are handled together. You will need to modify the `dateSplitter` function to always return a 4-digit year, and then handle the conversion to 2 digits in the `formatDate` function if necessary.